ndmg/stats/qa_fast.py

```diff
@@ -0,0 +1,207 @@
+            
+            if edge:
+                image = edge_map(image).data
+                image[image > 0] = max_val
+                image[image == 0] = min_val
+            
+            plt.xticks([])
+            plt.yticks([])
+            ax.spines['top'].set_visible(False)
+            ax.spines['right'].set_visible(False)
+            ax.spines['bottom'].set_visible(False)
+            ax.spines['left'].set_visible(False)
+            ax.imshow(atl, interpolation="none", cmap=cmaps[0], alpha=0.65)
+            ax.imshow(inim, interpolation="none", cmap=cmaps[1], alpha=0.65)
+            ax.imshow(image, interpolation="none", cmap=cmaps[2], alpha=0.6)
+            ax.imshow(
+                opaque_colorscale(
+                    cmaps[0], atl, alpha=0.65, vmin=min_val, vmax=max_val
+                )
+            )
+            ax.imshow(
+                opaque_colorscale(
+                    cmaps[1], inim, alpha=0.65, vmin=min_val, vmax=max_val
+                )
+            )
+            ax.imshow(
+                opaque_colorscale(
+                    cmaps[2], image, alpha=0.6, vmin=min_val, vmax=max_val
+                )
+            )
+            if idx == 3:
+                plt.plot(0, 0, "-", c='green', label='wm')
+                plt.plot(0, 0, "-", c='pink', label='gm')
+                plt.plot(0, 0, "-", c='blue', label='csf')
+                plt.legend(loc='upper right',fontsize=15,bbox_to_anchor=(1.5,1.5))
+            
+    foverlay.set_size_inches(12.5, 10.5, forward=True)
+    foverlay.tight_layout
+    return foverlay
+
+def get_min_max(data, minthr=2, maxthr=95):
+    """
+    data: regmri data to threshold.
+    """
+    min_val = np.percentile(data, minthr)
+    max_val = np.percentile(data, maxthr)
+    return (min_val.astype(float), max_val.astype(float))
+
+def reg_mri_pngs(
+    csf, gm, wm, outdir, loc=0, mean=False, minthr=2, maxthr=95, edge=False):
+    """
+    csf: directory where csf nifity file is loaded
+    gm: directory where gm nifity file is loaded
+    wm: directory where wm nifity file is loaded
+    outdir: directory where output png file is saved
+    fname: name of output file WITHOUT FULL PATH. Path provided in outdir.
+    """
+    gm_data = nb.load(gm).get_data()#csf
+    csf_data = nb.load(csf).get_data()#gm
+    wm_data = nb.load(wm).get_data()#wm
+
+    cmap1 = LinearSegmentedColormap.from_list("mycmap1", ["white", "magenta"])
+    cmap2 = LinearSegmentedColormap.from_list("mycmap2", ["white", "green"])
+    cmap3 = LinearSegmentedColormap.from_list("mycmap3", ["white", "blue"])
+
+    fig = plot_overlays(gm_data, csf_data, wm_data, [cmap1, cmap2, cmap3], minthr, maxthr, edge)
+
+    os.chdir(outdir.dirs['qa']['base'])
+    if 'preproc' not in os.listdir():
+        os.mkdir('preproc')
+    outdir.dirs['qa']['preproc'] = outdir.dirs['qa']['base'] + '/preproc/'
+
+    fig.savefig(outdir.dirs['qa']['preproc']+'fastOverlay.png', format="png")
```

Comment:
this block should be changed.
- `os.chdir` is absolutely unnecessary here, no need to change directories just to save a file
- just define a variable that has the path to the place that you want to save to once, then call `if not os.path.isdir(<preproc dir>)` to make it. 

Please explain why this is even necessary, though, since aren't you at a part of the pipeline where the directory should have always been made? Or am I wrong?